src/transform/src/predicate_pushdown.rs

```diff
@@ -459,95 +462,126 @@ impl PredicatePushdown {
                 equivalences,
                 ..
             } => {
-                expr::canonicalize::sort_dedup_equivalences(equivalences);
-                // Push down:
-                // 1) equality constraints supported by the same single
-                //    input.
-                // 2) equality constraints consisting of an expression
-                //    supported by a single input + a literal
+                // The goal is to either push or lift out of the join
+                // equivalences:
+                // 1) equivalences of the form `expr = literal`
+                // 2) equivalences of the form `expr1 = expr2`, where both
+                //    expressions come from the same single input.
+
+                expr::canonicalize::canonicalize_equivalences(equivalences);
 
                 let input_mapper = expr::JoinInputMapper::new(inputs);
-                // Predicates to push at each input.
+                // Predicates to push at each input, and to lift over the join.
```

Comment:
What does "lift over the join" mean? Above it is "lift out of the join" which does make sense, but actually I don't know why we would want to lift things rather than push them.